cart.rb

```diff
@@ -1,13 +1,31 @@
+# frozen_string_literal: true
+
 require 'pry'
 
 class Cart
+  attr_accessor :products_in_cart
+  attr_reader :cart_errors
+
   def initialize
-    @products = []
+    @products_in_cart = []
+    @cart_errors = []
   end
 
-  def add_product(item_id)
-    @products << item_id
+  def show_products
+    @products_in_cart.each do |product|
+      p product
+    end
   end
 
-  attr_reader :products
+  def add_product_to_cart(product)
```

Comment:
to tylko moje zdanie, nie wiem czy mam rację, ale jak wywołujesz `add_product_to_cart` na obiekcie `Cart` to jest to masło maślane, wiadomo że `to_cart`, wystarczyłoby `add_product`, tak samo `products` nie potrzebują `in_cart` a wręcz uważam że to spoko, że interfejs może być taki sam jak w `Shop` jest metoda `add_product` i w zależności czy wywołasz ją na Shop czy Cart, robi to samo